### ISSUE TITLE
Provide (safe) access to reflection protos in C#

### ISSUE THREAD

**Issue by jskeet** [protocolbuffers/protobuf#9425]
**What language does this apply to?**

C#

**Describe the problem you are trying to solve.**

Some APIs (e.g. [BigQuery Storage](https://github.com/googleapis/googleapis/blob/ab61c52324b39d1c8619a96ab19cc6c173b13f2c/google/cloud/bigquery/storage/v1/protobuf.proto#L39) use `DescriptorProto` for schema descriptions.

In C# we have access to reflection via the `MessageDescriptor` type, but there's no access to the underlying proto.

**Describe the solution you'd like**

`ToProto()` methods in all descriptor types, returning a clone of the underlying proto.

**Describe alternatives you've considered**

Access via .NET reflection :(

**Additional context**

I'll create a PR to implement this.

**Comment by jtattermusch** [comment]
Makes sense. I think historically we had to hide the *DescriptorProto  values since protobuf C# didn't have support for proto2 and the descriptor messages are proto2 messages. Not that we had proto2 support for a while, I guess we can provide access to the descriptor messages (if done in a safe way).
